akeneo-design-system/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,181 @@
+import React, {FC} from 'react';
+import styled, {css} from 'styled-components';
+import {getColor, AkeneoThemedProps} from '../../theme';
+
+type PaginationProps = {
+  /**
+   * The current page number (starting à 1).
+   */
+  currentPage: number;
+
+  /**
+   * The total number of pages.
+   */
+  itemsTotal: number;
+
+  /**
+   * The number of items per page. Default to 25.
+   */
+  itemsPerPage?: number;
+
+  /**
+   * Handler called when a pagination item is clicked.
+   */
+  onClick: (page: number) => void;
+};
+
+const MAX_PAGINATION_ITEMS_WITHOUT_SEPARATOR = 4;
+const PAGINATION_SEPARATOR = '...';
+
+const Pagination: FC<PaginationProps> = ({currentPage, itemsTotal, itemsPerPage = 25, onClick}) => {
+  if (itemsPerPage <= 0) {
+    return <></>;
+  }
+
+  const numberOfPages = Math.ceil(itemsTotal / itemsPerPage);
+
+  if (numberOfPages <= 1 || currentPage > numberOfPages) {
+    return <></>;
+  }
+
+  const pages = computePages(currentPage, numberOfPages);
+
+  return (
+    <PaginationContainer>
+      {pages.map((page: number | string, index: number) => {
+        return (
+          <PaginationItem currentPage={page === currentPage} key={index} onClick={onClick}>
+            {page}
+          </PaginationItem>
+        );
+      })}
+    </PaginationContainer>
+  );
+};
+
+const PaginationContainer = styled.div`
+  height: 44px;
+  margin: 10px 0 10px 0;
+  align-items: center;
+  display: flex;
+  justify-content: center;
+`;
+
+type PaginationItemProps = {
+  currentPage: boolean;
+  onClick: (page: number) => void;
+};
+
+const PaginationItem: FC<PaginationItemProps> = ({currentPage, children, onClick}) => {
+  const title = `No. ${children as string}`;
+
+  return (
+    <PaginationItemContainer
+      onClick={() => (children == PAGINATION_SEPARATOR ? null : onClick(children as number))}
+      data-testid="paginationItem"
+      title={children != PAGINATION_SEPARATOR ? title : ''}
+      disabled={children == PAGINATION_SEPARATOR}
+      currentPage={currentPage}
+    >
+      {children}
+    </PaginationItemContainer>
+  );
+};
+
+const currentPaginationItemMixin = css`
+  border: 1px ${getColor('brand', 100)} solid;
+  color: ${getColor('brand', 100)};
+`;
+
+const otherPaginationItemMixin = css`
+  border: 1px ${getColor('grey', 80)} solid;
+  color: ${getColor('grey', 100)};
+`;
+
+const disabledMixin = css`
+  cursor: default;
+  :hover {
+    background-color: ${getColor('white')};
+  }
+`;
+
+const PaginationItemContainer = styled.span<AkeneoThemedProps & PaginationItemProps & {disabled: boolean}>`
+  ${props => (props.currentPage ? currentPaginationItemMixin : otherPaginationItemMixin)}
```

Comment:
pro tips: if you want, you can to this:
```suggestion
  ${({currentPage}) => (currentPage ? currentPaginationItemMixin : otherPaginationItemMixin)}
```